### PR TITLE
feat: Add copy link to clipboard button(rupali-codes#362)

### LINKED ISSUE #362: [Feature] Add copy link to clipboard button
### Description

Hey, I just noticed that adding `copy link to clipboard` would be a great user experience, as user will be able to copy the link share among their friends without opening it.

### Specific Requirements
* Add `copy to clipboard` icon beside title in the card on the right side
* Also, add the icon in `popup` on the right side
* When clicked on the icon, the link of the respective resource should be copied to user's clipboard

### Screenshots

**For Card**

image

**For popup**

image

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sloth30799**
## Fixes Issue
Closes #362 

## Changes proposed
Added CopyToClipboard component and hook

## Screenshots
Web capture_20-3-2023_182833_localhost
Web capture_20-3-2023_182820_localhost

### CODE REVIEW

**Comment by rupali-codes** [comment]
Looks nice, can we use this icon _FaRegCopy_, and also update the color to _violet_

image

**Review by ujjawaltyagii — COMMENTED** [review]
Looks nice to me !

File: components/popup/index.tsx
Comment (rupali-codes) [inline components/popup/index.tsx:1]: the cursor is still normal when hovering on copy icon, please add `cursor-pointer`

Comment (sloth30799) [reply components/popup/index.tsx:1]: added cursor pointer

**Review by rupali-codes — APPROVED** [review]
thanks, lgtm

**Review by ujjawaltyagii — APPROVED** [review]
with new icon it's perfect ! nicely done 👍

**Pull request merged**